specification/eventgrid/data-plane/Microsoft.Communication/stable/2018-01-01/AzureCommunicationServices.json

```diff
@@ -1,745 +1,886 @@
+            "description": "Schema of the Data property of an EventGridEvent for a Microsoft.Communication.CallStarted event.",
+            "type": "object",
+            "allOf": [
+                {
+                    "$ref": "#/definitions/AcsCallingEventBaseProperties"
+                }
+            ]
+        },
+        "AcsCallEndedEventData": {
+            "description": "Schema of the Data property of an EventGridEvent for a Microsoft.Communication.CallEnded event.",
+            "type": "object",
+            "allOf": [
+                {
+                    "$ref": "#/definitions/AcsCallingEventBaseProperties"
+                }
+            ],
+            "properties": {
+                "durationOfCall": {
+                    "description": "The duration of the call",
+                    "type": "number",
+                    "format": "double"
+                }
+            }
+        },
+        "AcsCallParticipantAddedEventData": {
+            "description": "Schema of the Data property of an EventGridEvent for a Microsoft.Communication.CallParticipantAdded event.",
+            "type": "object",
+            "allOf": [
+                {
+                    "$ref": "#/definitions/AcsCallingParticipantEventBaseProperties"
+                }
+            ]
+        },
+        "AcsCallParticipantRemovedEventData": {
+            "description": "Schema of the Data property of an EventGridEvent for a Microsoft.Communication.CallParticipantRemoved event.",
+            "type": "object",
+            "allOf": [
+                {
+                    "$ref": "#/definitions/AcsCallingParticipantEventBaseProperties"
+                }
+            ]
+        },
+        "AcsCallingParticipantEventBaseProperties": {
+            "description": "Schema of common properties of all Calling Participant events",
+            "type": "object",
+            "allOf": [
+                {
+                    "$ref": "#/definitions/AcsCallingEventBaseProperties"
+                }
+            ],
+            "properties": {
+                "userCommunicationIdentifier": {
+                    "description": "The communication identifier of the user who was added to the call",
+                    "$ref": "../../../../../communication/data-plane/Common/stable/2021-03-07/common.json#/definitions/CommunicationIdentifierModel"
+                },
+                "participantId": {
+                    "description": "The id of the user who was added",
+                    "type": "string"
+                },
+                "displayName": {
+                    "description": "The display name of the user who was added",
+                    "type": "string"
+                },
+                "endpointType": {
+                    "description": "The type of the user's endpoint added",
+                    "type": "string"
+                }
+            }
+        },
+        "AcsCallingEventBaseProperties": {
```

Comment:
Are customers expected to use this Base event type, or is it just being added to avoid duplicating common properties in the different events?

Reply:
to avoid duplicating common properties